src/librustdoc/html/format.rs

```diff
@@ -152,159 +152,172 @@ fn comma_sep<T: fmt::Display>(items: impl Iterator<Item = T>) -> impl fmt::Displ
     })
 }
 
-crate fn print_generic_bounds(bounds: &[clean::GenericBound]) -> impl fmt::Display + '_ {
+crate fn print_generic_bounds<'a>(
+    bounds: &'a [clean::GenericBound],
+    cache: &'a Cache,
+) -> impl fmt::Display + 'a {
```

Comment:
I think this can be replaced with:

```suggestion
crate fn print_generic_bounds(
    bounds: &[clean::GenericBound],
    cache: &Cache,
) -> impl fmt::Display + '_ {
```

Reply:
This whole file is just lifetime dark magic. And no, unfortunately it cannot. It was a real nightmare to update all these functions... :crying_cat_face:

Reply:
Huh, that's weird. I thought it would be supported by the lifetime elision rules. Oh well :)

Reply:
No, it's very tricky because of this function:

```rust
impl<'a> WhereClause<'a> {
    crate fn print<'b>(&'b self, cache: &'b Cache) -> impl fmt::Display + 'b {
```

Everything is because of this specific function. And it was worse before I made the changes so that the closure didn't require `'static` lifetime.